presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/RecordingHiveMetastore.java

```diff
@@ -73,6 +74,7 @@
     private final Cache<HiveTableName, PartitionStatistics> tableStatisticsCache;
     private final Cache<Set<HivePartitionName>, Map<String, PartitionStatistics>> partitionStatisticsCache;
     private final Cache<String, List<String>> allTablesCache;
+    private final Cache<List<String>, List<String>> tablesWithParameterCache;
```

Comment:
same comment about cache key here

Reply:
Creating a class (must be JSON serializable) would improve readability of the generated recordings.